src/cocotb/share/include/gpi.h

```diff
@@ -142,6 +142,8 @@ GPI_EXPORT const char *gpi_get_simulator_version(void);
 GPI_EXPORT gpi_sim_hdl gpi_get_root_handle(const char *name);
 GPI_EXPORT gpi_sim_hdl gpi_get_handle_by_name(gpi_sim_hdl parent,
                                               const char *name);
+GPI_EXPORT gpi_sim_hdl gpi_get_handle_by_name_native(gpi_sim_hdl parent,
```

Comment:
I think instead maybe having an argument to `gpi_get_handle_by_name` that you can specify `NATIVE` or `AUTO`, etc would be better. That way we can add additional discovery methods latter if need be.

Reply:
I agree this would be better, I just did'nt really dare to modify the existing signature.

As for NATIVE or AUTO: how would we specify these on the C++ and Python ? Defines? Enum?
Not that experienced in designing Python-C-intrerfaces so some guidance there would be appreciated.

Reply:
enums in C, see some examples a few line above in gpi.h. `IntEnum`s in Python, see `_GPISetAction` in handle.py as an example. There's a way to get it all to work together, but we aren't doing that yet, so don't feel you need to be the first.

Reply:
pushed new version doing that.